components/FriendPagePreview.js

```diff
@@ -70,33 +88,62 @@ export default function FriendPagePreview({ friend }) {
           height={180}
         />
       </div>
-
       {/* <p>Contact:</p> */}
+
       <StyledContactList>
-        {Object.entries(friend.contactOptions).map(([key, value]) => (
-          <StyledList key={key}>
-            <StyledButton href={value}>
-              {key}{" "}
-              {/* : <br />
+        {friend.contactOptions &&
+          friend.contactOptions.map((options) => (
+            <StyledList key={options.name}>
+              <StyledButton href={options.number}>
+                {options.name}
+
+                {/* : <br />
               {value} */}
-            </StyledButton>
-          </StyledList>
-        ))}
+              </StyledButton>
+            </StyledList>
+          ))}
       </StyledContactList>
       <StyledInfoList>
-        <p>Birthday: {friend.birthday}</p>
-        <p>Status: {friend.friendship_status}</p>
-        <p>City: {friend.city}</p>
+        <p>
+          Birthday: <b>{friend.birthday}</b>
+        </p>
+        <p>
+          Status: <b>{friend.friendship_status}</b>
+        </p>
+        <p>
+          City: <b>{friend.city}</b>
+        </p>
         <p>Hobbies:</p>
         <ul>
-          {friend.hobbies.map((hobby, index) => (
-            <li key={index}>{hobby}</li>
-          ))}
+          {friend.hobbies &&
+            friend.hobbies.map((hobbie, index) => (
+              <li key={index}>
+                <b>{hobbie}</b>
+              </li>
+            ))}
         </ul>
       </StyledInfoList>
       <StyledNotice>
-        <p>Notice:</p>
+        <p>Your Notice:</p>
       </StyledNotice>
+      <StyledButtonContact onClick={handleClick}>
+        <div>
+          <p>Contact Options </p>
+        </div>
+        {value && (
+          <div>
+            <ul>
+              {friend.contactOptions.map((options) => (
+                <list key={options.name}>
+                  <li href={options.number}>
+                    {options.name} : <b>{options.number || options.address}</b>
+                  </li>
+                </list>
+              ))}
+            </ul>
+          </div>
+        )}
+      </StyledButtonContact>
```

Comment:
```suggestion
      <StyledButtonContact onClick={handleClick}> Contact </StyledButtonContact>
      {isDisplayed && (
        <ul>
          {friend.contactOptions.map((options) => (
            <list key={options.name}>
              <li href={options.number}>
                {options.name} : <b>{options.number || options.address}</b>
              </li>
            </list>
          ))}
        </ul>
      )}
```

To include block elements inside of button elements is not good HTML. [See caninclude](https://caninclude.glitch.me/caninclude?child=div&parent=button)
Instead try to rewrite this part like I suggest.